query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -263,9 +277,13 @@ private String convertExpression2String(
       Expression expression, Builder paramsBuilder, ExecutionContext executionContext) {
     switch (expression.getValueCase()) {
       case COLUMNIDENTIFIER:
-        String logicalColumnName = expression.getColumnIdentifier().getColumnName();
         // this takes care of the Map Type where it's split into 2 columns
-        List<String> columnNames = viewDefinition.getPhysicalColumnNames(logicalColumnName);
+        List<String> columnNames =
+            viewDefinition.getPhysicalColumnNames(getLogicalColumnName(expression));
+        return joiner.join(columnNames);
+      case ATTRIBUTEEXPRESSION:
+        // this takes care of the Map Type where it's split into 2 columns
+        columnNames = viewDefinition.getPhysicalColumnNames(getLogicalColumnName(expression));
```

Comment:
Bad practice to share variables between cases. I'm actually not even sure what java does in this case since control flow never hits the declaration on line 281 - probably doesn't care, but I'd be curious. Either way, best avoided. Just inline columnNames inside the join fn.

Reply:
Java gives error if we declare the same variable in different `CASE` statements. 
Quoted from online sources : 
"
That's because a case doesn't create a scope. So, both the variables in the 2 cases are in the same scope. If you want to do this, you can add braces for each case, which will create a new scope for each case.
"
Combined both the cases as of now.

Reply:
Yes, you can't redeclare the same variable between case statements - but there's no need for a variable here, it's only used on the following line and doesn't add clarity.

Reply:
resolved